.github/workflows/test_on_push.yml

```diff
@@ -47,6 +47,11 @@ jobs:
         with:
           python-version: ${{ matrix.python-version }}
 
+      - name: Set up Git
+        run: |
+          git config --global user.name "pybamm user"
+          git config --global user.email "pybammuser@pybamm.org"
+
```

Comment:
Nitpicky comment: could you move this step so that it is just before where the tests are run? That makes it a little clearer as to what the step is required for – and that the `uv` and `nox` installation steps are not related to this.